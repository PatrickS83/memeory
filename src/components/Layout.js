import React from 'react';
import PropTypes from 'prop-types';
import Navbar from './Navbar';
import Footer from './Footer';

const Layout = props => (
  <div className="layout">
    <Navbar />
    {props.children}
    <Footer />
  </div>
);

Layout.propTypes = {
  children: PropTypes.node.isRequired
};

export default Layout;
