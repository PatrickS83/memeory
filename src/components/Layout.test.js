import React from 'react';
import { shallow } from 'enzyme';
import Layout from './Layout';

describe('<Layout/>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <Layout>
        <p>TestChild</p>
      </Layout>
    );
    expect(wrapper.length).toEqual(1);
  });

  it('renders the Layout as saved in snapshot', () => {
    const wrapper = shallow(
      <Layout>
        <p>TestChild</p>
      </Layout>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the children prop', () => {
    const wrapper = shallow(
      <Layout>
        <p>TestChild</p>
      </Layout>
    );
    expect(wrapper.find('p').length).toBe(1);
  });
});
