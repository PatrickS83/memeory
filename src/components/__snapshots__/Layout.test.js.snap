// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout/> renders the Layout as saved in snapshot 1`] = `
<div
  className="layout"
>
  <Navbar />
  <p>
    TestChild
  </p>
  <Footer />
</div>
`;
